noxfile.py

```diff
@@ -162,7 +162,14 @@ def pip_compile(session: nox.Session, req: str):
 
     # Use --upgrade by default unless a user passes -P.
     args = list(session.posargs)
-    if not any(
+
+    # Support a custom --check flag to fail if pip-compile made any changes
+    # so we can check that that lockfiles are in sync with the input (.in) files.
+    check_mode = "--check" in args
+    if check_mode:
+        # Remove from args, as pip-compile doesn't actually support --check.
+        args.remove("--check")
```

Comment:
Not to be annoying about it but I've been reading through this thread: https://github.com/jazzband/pip-tools/issues/882

It sounds like there are some folks who want to achieve a similar result. I've noticed one project that does `pip-compile --quiet && git diff-index --quiet HEAD --` to detect changes and then error exit. I wonder if an approach like that might fit the bill here too? Do the `dry-run` and `quiet` options provide most of what we need, if not exact?

Reply:
That works nicely in shell, but nox doesn't run a shell. The implementation here simply checks if `git diff` produces any output (which nox's `Session.run()` makes pretty easy to do) and then prints out an informative message.

Reply:
Right, unfortunately, pip-compile does not have a built-in way to do this, so I figured adding our own `--check` flag would be the best approach.